src/main/java/games/chess/actions/EnPassant.java

```diff
@@ -0,0 +1,136 @@
+package games.chess.actions;
+
+import core.AbstractGameState;
+import core.actions.AbstractAction;
+import core.components.Component;
+import games.chess.ChessGameState;
+import games.chess.components.ChessPiece;
+import java.lang.System;
+import java.util.Objects;
+
+/**
+ * <p>Actions are unit things players can do in the game (e.g. play a card, move a pawn, roll dice, attack etc.).</p>
+ * <p>Actions in the game can (and should, if applicable) extend one of the other existing actions, in package {@link core.actions}.
+ * Or, a game may simply reuse one of the existing core actions.</p>
+ * <p>Actions may have parameters, so as not to duplicate actions for the same type of functionality,
+ * e.g. playing card of different types (see {@link games.sushigo.actions.ChooseCard} action from SushiGo as an example).
+ * Include these parameters in the class constructor.</p>
+ * <p>They need to extend at a minimum the {@link AbstractAction} super class and implement the {@link AbstractAction#execute(AbstractGameState)} method.
+ * This is where the main functionality of the action should be inserted, which modifies the given game state appropriately (e.g. if the action is to play a card,
+ * then the card will be moved from the player's hand to the discard pile, and the card's effect will be applied).</p>
+ * <p>They also need to include {@link Object#equals(Object)} and {@link Object#hashCode()} methods.</p>
+ * <p>They <b>MUST NOT</b> keep references to game components. Instead, store the {@link Component#getComponentID()}
+ * in variables for any components that must be referenced in the action. Then, in the execute() function,
+ * use the {@link AbstractGameState#getComponentById(int)} function to retrieve the actual reference to the component,
+ * given your componentID.</p>
+ */
```

Comment:
This boiler plate makes sense in the gametemplate package - but when you implement concrete classes you shoudl replace it with doc that makes sense to the implementation (i.e. here, one or two lines about En Passant...or even just delete it to avoid duplicated cruft)